engine/tests/classes/Elgg/TestableHook.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace Elgg;
+
+class TestableHook {
+
+	/**
+	 * @var string
+	 */
+	public $name;
+
+	/**
+	 * @var string
+	 */
+	public $type;
+
+	/**
+	 * @var callable
+	 */
+	public $handler;
+
+	/**
+	 * @var int
+	 */
+	public $priority = 500;
+
+	/**
+	 * @var BaseTestCase
+	 */
+	protected $test_case;
+
+	/**
+	 * @var int
+	 */
+	protected $calls = 0;
+
+	/**
+	 * @var Hook
+	 */
+	protected $before_state;
+
+	/**
+	 * @var mixed
+	 */
+	protected $result;
+
+	/**
+	 * @var Hook
+	 */
+	protected $after_state;
+
+	/**
+	 * Register a new testing hook handler
+	 *
+	 * @param BaseTestCase $test_case Test case
+	 *
+	 * @return TestableHook
+	 */
+	public function register(BaseTestCase $test_case) {
+		$this->test_case = $test_case;
+
+		elgg_register_plugin_hook_handler($this->name, $this->type, [$this, 'handler'], $this->priority);
+
+		return $this;
+	}
+
+	/**
+	 * Unregister handler
+	 * @return void
+	 */
+	public function unregister() {
+		elgg_unregister_plugin_hook_handler($this->name, $this->type, [$this, 'handler']);
+	}
+
+	/**
+	 * Handler hook call
+	 *
+	 * @param Hook $hook Hook
+	 *
+	 * @return mixed
+	 */
+	public function handler(\Elgg\Hook $hook) {
+		$this->calls++;
+		$this->before_state = $hook;

```

Comment:
Before and after will reference the same object. You need to clone particular properties of the hook to capture differences.

Reply:
Can be done later. I didn't need it for this test, and I can't see it anymore. Finishing one last thing, and I am done with Elgg for a while.